Add rocket.chat

pierreozoux opened the pull request.

geekgonecrazy commented on stable/rocketchat/templates/deployment.yaml:
```diff
@@ -0,0 +1,61 @@
+              name: {{ template "fullname" . }}
+              key: mongo-uri
+        {{ else }}
+          value: mongodb://{{ template "mongodb.fullname" . }}:27017/meteor
```
To increase consistency with other Rocket.Chat documentation maybe use rocketchat as the db name here?

geekgonecrazy commented on stable/rocketchat/templates/ingress.yaml:
```diff
@@ -0,0 +1,20 @@
+  name: {{ template "fullname" . }}
+  annotations:
+    kubernetes.io/tls-acme: "true"
+    kubernetes.io/ingress.class: nginx
```
This automatically assumes they have an nginx ingress running.  Is this safe?

linki replied:
It should be commented out by default. Similar to https://github.com/kubernetes/charts/blob/e80d1cfb8857cc71096b3b8a4638a789b378db88/stable/grafana/values.yaml#L23

This will allow the Chart to work by default. People that use the nginx-based ingress controller can either opt-in by setting this to "nginx" or they configure their cluster/ingress-controller to pick up Ingresses without annotations, too (like GKE does).

linki commented:
@pierreozoux can you have another look at this?

@geekgonecrazy Thanks for your thoughts.

pierreozoux commented:
@linki I addressed all the comments already (I didn't comment, I just fixed it to avoid noise).

But yeah, thanks @geekgonecrazy for your comments, I fixed them all :)

Let me know if there is anything more needed?

linki commented on stable/rocketchat/.helmignore:
```diff
@@ -0,0 +1 @@
+.git
```
This is not the standard `.gitignore`. Let's wait on the outcome of https://github.com/kubernetes/charts/pull/485/files#r106305261

Looks like we prefer the "canonical" helmignore from now on but many charts still have just this. It probably makes more sense to sync them up in a separate PR across all charts.

pierreozoux commented:
Is there anything I can do to get this merged?

engelgabriel commented:
The Rocket.Chat team is looking forward for this!

geekgonecrazy commented:
@pierreozoux one thing to maybe add here is the environment variables to populate the INSTANCE_IP variable with the pod ip.

Here is a snippet I have for adding this to the schema:
```
name: INSTANCE_IP
valueFrom:
  fieldRef:
    fieldPath: status.podIP
```

This way if scaled the instances can know how to talk to each other over the overlay network

pierreozoux commented:
@geekgonecrazy hum, do you have the link to the documentation?

If the person uses S3, it would scale automatically out of the box.
Let's think about this INSTANCE_IP (which is a better way I guess) for another PR, once this got merged?

geekgonecrazy commented:
@pierreozoux 
Here's a snippet directly out of one of our deployments:
testdeploy.yaml
<details>

```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  labels:
    DeploymentDomain: testdeploy.rocket.chat
    DeploymentName: testdeploy
  name: testdeploy
  namespace: <namespace here>
spec:
  replicas: 2
  revisionHistoryLimit: 1
  selector:
    matchLabels:
      DeploymentDomain: testdeploy.rocket.chat
      DeploymentName: testdeploy
  strategy:
    rollingUpdate:
      maxSurge: 1
      maxUnavailable: 1
    type: RollingUpdate
  template:
    metadata:
      creationTimestamp: null
      labels:
        DeploymentDomain: testdeploy.rocket.chat
        DeploymentName: testdeploy
    spec:
      containers:
      - env:
        - name: INSTANCE_IP
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: status.podIP
        - <another 10-20 ommited for briefness>
        image: rocketchat/rocket.chat:0.55.1
        imagePullPolicy: IfNotPresent
        name: testdeploy
        ports:
        - containerPort: 3000
          protocol: TCP
```

</details>
Its for sure something that could be added in later.

viglesiasce commented:
@k8s-bot ok to test

viglesiasce commented on stable/rocketchat/templates/deployment.yaml:
```diff
@@ -0,0 +1,65 @@
+{{- if include "host" . -}}
```
What is this line trying to do? Shouldnt it be:

{{- if .Values.host -}}

viglesiasce replied:
The linter is also complaining about it and I the pod is crashing with the following when it is not set:

```
/app/bundle/programs/server/boot.js:356
}).run();
   ^

Error: $ROOT_URL, if specified, must be an URL
    at Error (native)
    at packages/meteor.js:1218:13
    at packages/meteor.js:1233:4
    at packages/meteor.js:1380:3
    at /app/bundle/programs/server/boot.js:303:34
    at Array.forEach (native)
    at Function._.each._.forEach (/app/bundle/programs/server/node_modules/underscore/underscore.js:79:11)
    at /app/bundle/programs/server/boot.js:128:5
    at /app/bundle/programs/server/boot.js:352:5
    at Function.run (/app/bundle/programs/server/profile.js:510:12)
```

If this is a required field look at what ghost and gitlab-ce have done for this use case.

geekgonecrazy commented:
@pierreozoux do you need any help getting these changes made?  Would be glad to help out

pierreozoux commented:
@viglesiasce thanks I addressed your comment.

pierreozoux commented:
Is there anything else I can do to help?

geekgonecrazy commented:
@viglesiasce / @linki we would love to see this progress forward.  Is there anything you guys need on this to get it merged?

unguiculus commented on stable/rocketchat/Chart.yaml:
```diff
@@ -0,0 +1,20 @@
+sources:
+- https://github.com/RocketChat/Docker.Official.Image/
+maintainers:
+- name: Rocket.Chat
```
Use Github username.

unguiculus commented on stable/rocketchat/templates/deployment.yaml:
```diff
@@ -0,0 +1,65 @@
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "fullname" . }}
```
`app` label should be `{{ template "name" . }}`. Apply everywhere.

unguiculus commented on stable/rocketchat/templates/svc.yaml:
```diff
@@ -0,0 +1,16 @@
+    port: 3000
+    targetPort: http
+  selector:
+    app: {{ template "fullname" . }}
```
After changing the `app` label as mentioned above, the selector should be as follows:

```yaml
app: {{ template "name" . }}
release: "{{ .Release.Name }}"
```

unguiculus commented on stable/rocketchat/values.yaml:
```diff
@@ -0,0 +1,85 @@
+## Configure resource requests and limits
+## ref: http://kubernetes.io/docs/user-guide/compute-resources/
+##
+resources:
```
We've started recommending not to specify default resources.

pierreozoux committed:
Add rocket.chat

pierreozoux commented:
@unguiculus Thanks for your comment, I addressed them.

unguiculus commented on stable/rocketchat/Chart.yaml:
```diff
@@ -0,0 +1,20 @@
+sources:
+- https://github.com/RocketChat/Docker.Official.Image/
+maintainers:
+- name: RocketChat
```
Name should be a Github user.

geekgonecrazy replied:
@unguiculus this is the org name.  Do we need to add individual maintainers listed here?

geekgonecrazy replied:
@pierreozoux I can provide a list from Rocket.Chat that would handle any issues that would arise as well as of course your self.

geekgonecrazy replied:
@unguiculus I notice many of the bitnami created ones refer to their org name on github and add an email address.  But i've gone ahead and added two additional maintainers by github username.

unguiculus commented on stable/rocketchat/templates/NOTES.txt:
```diff
@@ -0,0 +1,22 @@
+
+You can easily connect to the remote instance from your browser. Forward the webserver port to localhost:8888
+
+- kubectl port-forward --namespace {{ .Release.Namespace }} $(kubectl get pods --namespace {{ .Release.Namespace }} -l app={{ template "fullname" . }} -o jsonpath='{ .items[0].metadata.name }') 8888
```
Fix label selectors:
`-l app="{{ template "fullname" . }},release={{ .Release.Name }}"`

geekgonecrazy replied:
Fixed, also added these labels to the actual pod level so the command would work in the first place :)

unguiculus commented on stable/rocketchat/templates/deployment.yaml:
```diff
@@ -0,0 +1,65 @@
+{{- if default "" .Values.host }}
```
This is empty by default. When I install the chart with defaults only MongoDB comes up. It should be possible to install a chart with default values.

geekgonecrazy replied:
Fixed

geekgonecrazy committed:
Add additional maintainers beyond the github org

geekgonecrazy committed:
Fix mongo dependency

geekgonecrazy committed:
Improved Labels, Chart Defaults, Notes
* Add labels to pod for proper kubectl selection
* Allow chart to be installed with no values set
* Improved output notes

geekgonecrazy commented:
Bump to verify CLA

geekgonecrazy committed:
add helm chart to suggested install command

geekgonecrazy committed:
Add deploy platform environment variable

geekgonecrazy committed:
Add Instance IP to pod env variable to make scaling easier

geekgonecrazy committed:
Switch app name to use app's name

geekgonecrazy committed:
Update to latest Rocket.Chat version

geekgonecrazy committed:
Remove else from ROOT_URL, its not needed

geekgonecrazy commented:
Helping @pierreozoux get things finished here.

@unguiculus @viglesiasce I think i've now addressed all feedback. 

* Added maintainers.  I went ahead and left our org and email for contact, and added two maintainers to the list.
* Fixed Notes so the output will suggest how to get site going at http://chat.yourdomain.com.  Fixed a few selectors in the queries it says to run as well as the port forward port
* Now can be installed with as little as: `helm install stable/rocketchat` brings up mongodb and rocket.chat

I've also signed the CLA

geekgonecrazy commented:
@viglesiasce @unguiculus @linki would any of you guys be able to take a look at this?

viglesiasce commented:
@geekgonecrazy I'll take a look today.

geekgonecrazy commented:
@viglesiasce thank you!  I've also joined in the slack #helm-users channel and glad to chat and address what ever concerns if any still remain.

viglesiasce commented:
Worked a treat! Thanks!

geekgonecrazy commented:
Perfect!  Thanks @viglesiasce !

pierreozoux commented:
\o/ Thanks!

sathieu commented:
Hello,

It is not on https://kubeapps.com/. Can someone push it?

sathieu commented:
and there is no README...

unguiculus commented:
@prydonius Could you check why the chart is not listed?

geekgonecrazy commented:
Would be glad to help make what ever changes are needed to make it show up.

sathieu commented:
Maybe this is the README missing?
